tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/operations/ProjectTestOperations.java

```diff
@@ -139,6 +131,62 @@ public static void openProjectComponentYMLFileInTextEditorUsingContextMenu(Strin
 	        throw new RuntimeException("The 'idf_component.yml' file was not found in the 'main' subfolder.");
 	    }
 	}
+	
+	public static void findProjectCleanedFilesInBuildFolder(String projectName, SWTWorkbenchBot bot) 
+	{
+	    SWTBotTreeItem projectItem = fetchProjectFromProjectExplorer(projectName, bot);
+	    if (projectItem != null) {
+	        projectItem.select();
+	        projectItem.expand();
+	        projectItem.getNode("build").expand();
+
+	        int maxAttempts = 2;
+	        for (int attempt = 0; attempt <= maxAttempts; attempt++) {
+	            SWTBotTreeItem fileToBeAbsentItem = isFileAbsent(projectItem.getNode("build"), ".elf");
+	            SWTBotTreeItem fileToFindItem = findTreeItem(projectItem.getNode("build"), "bootloader");
+	            if (fileToFindItem != null && fileToBeAbsentItem != null) {
+	                return;
+	            }
+
+	            else {
+	                try {
+	                    Thread.sleep(3000);
+	                } catch (InterruptedException e) {
+	                    e.printStackTrace();
+	                }
+	            }
+	        }
+	        throw new RuntimeException("The file was not deleted from the 'build' subfolder.");
+	    }
+	}
+	
+	public static void findProjectFullCleanedFilesInBuildFolder(String projectName, SWTWorkbenchBot bot) 
+	{
+	    SWTBotTreeItem projectItem = fetchProjectFromProjectExplorer(projectName, bot);
+	    if (projectItem != null) {
+	        projectItem.select();
+	        projectItem.expand();
+	        projectItem.getNode("build").expand();
+
+	        int maxAttempts = 2;
+	        for (int attempt = 0; attempt <= maxAttempts; attempt++) {
+	            SWTBotTreeItem fileToBeAbsentItem = isFileAbsent(projectItem.getNode("build"), ".elf");
+	            SWTBotTreeItem fileToBeAbsentItem1 = isFileAbsent(projectItem.getNode("build"), "bootloader");
+	            if (fileToBeAbsentItem != null && fileToBeAbsentItem1 != null) {
+	                return;
+	            }
+
+	            else {
+	                try {
+	                    Thread.sleep(3000);
+	                } catch (InterruptedException e) {
+	                    e.printStackTrace();
+	                }
```

Comment:
There is a more robust way to ensure that a clean command is finished. We can check if a clean job is still running via Job Manager and check the file for existence after the job is finished